core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -809,9 +815,11 @@ public MergeWriterNode plan(Merge merge)
                         rewritten = new CoalesceExpression(rewritten, new Cast(failFunction(metadata, session, INVALID_ARGUMENTS, "Assigning NULL to non-null MERGE target table column " + columnName), toSqlType(columnSchema.getType())));
                     }
                     rowBuilder.add(rewritten);
+                    checkConstraintsAssignments.put(field, rewritten);
```

Comment:
This makes sense only when `if (!constraints.isEmpty())` right?

Same for line `822`

Reply:
Yes. It's done similar as for `UPDATE`. Do you think that we need here additional complication?